File: Algorithm/src/main/java/Acwing/Ac828_模拟栈/main.go
package main

import (
	"bufio"
	"os"
	"strconv"
)

var scanner = bufio.NewScanner(os.Stdin)
var writer = bufio.NewWriter(os.Stdout)

type any = interface{}

func nextInt() int {
	scanner.Scan()
	ans, _ := strconv.Atoi(scanner.Text())
	return ans
}

func nextStr() string {
	scanner.Scan()
	return scanner.Text()
}
func init() {
	scanner.Split(bufio.ScanWords)
}

type stack struct {
	data [100005]any
	top  int
}

func (s *stack) push(x any) {
	s.top++
	s.data[s.top] = x
}

func (s *stack) pop() {
	s.top--
}

func (s *stack) empty() bool {
	return s.top == -1
}

func (s *stack) query() any {
	return s.data[s.top]
}
func main() {
	n := nextInt()
	stack := stack{top: -1}
	for i := 0; i < n; i++ {
		command := nextStr()
		switch command {
		case "push":
			num := nextInt()
			stack.push(num)
			break
		case "pop":
			stack.pop()
			break
		case "empty":
			if stack.empty() {
				writer.Write([]byte("YES\n"))
				//fmt.Println("YES")
			} else {
				writer.Write([]byte("NO\n"))
				//fmt.Println("NO")
			}

			break
		case "query":
			writer.Write([]byte(strconv.Itoa(stack.query().(int)) + "\n"))
			//fmt.Println(stack.query())
			break
		default:
			break
		}
	}
	writer.Flush()
}

File: Algorithm/src/main/java/Acwing/Ac828_模拟栈/Main.java
package Acwing.Ac828_模拟栈;

import java.io.*;
class Stack<T> {
    T[] arr;
    int top;

    public Stack() {
        this.top = -1;
        arr = (T[]) new Object[100005];
    }

    public void pop(){
        this.top--;
    }

    public void push(T t){
        this.top++;
        arr[this.top] = t;
    }

    public T query(){
        return arr[this.top];
    }

    public boolean isEmpty(){
        return this.top == -1;
    }
}

public class Main {
    static BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    static StreamTokenizer st = new StreamTokenizer(in);
    static int nextInt() throws IOException {
        st.nextToken();
        return (int) st.nval;
    }

    static String nextStr() throws IOException {
        st.nextToken();
        return st.sval;
    }

    static PrintWriter out = new PrintWriter(System.out);

    public static void main(String[] args) throws IOException {
        int n = nextInt();
        Stack<Integer> stack = new Stack<>();
        while (n-->0){
            String command = nextStr();
            switch (command){
                case "empty":
                    out.println(stack.isEmpty()?"YES":"NO");
                    break;
                case "query":
                    out.println(stack.query());
                    break;
                case "push":
                    stack.push(nextInt());
                    break;
                case "pop":
                    stack.pop();
                    break;
                default:
                    break;
            }
        }
        out.flush();
    }
}
